### ISSUE TITLE
IDEATION: Text formatting issues in idea.

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1310]
Screen recording here showing issues with trying to make headings bold:

https://www.loom.com/share/81aa5a8637914ae9803f52ed1a1d7aac?sid=6e4b0dc8-7d30-4d3b-9cb2-07ff6bdbfc6a

Though on posting it I see no formatting is possible currently. Is it possible to at least have paragraph spacing and the basic formatting that's offered in the text field? (Or I guess is just a bug). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/20c3e4b5-4458-4721-a5d3-afc2d14548d6)

**Comment by BeccaMelhuish** [comment]
Seems to be generally something up with the formatting of ideas actually (short as well as long) 

As this was a bold line then 4 lines of bullet points, and comes out like this when posted: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/65820dd7-4ac7-41d2-94a2-c561039ae1ac)

**Comment by BeccaMelhuish** [comment]
User on iPhone Safari found a similar thing when trying to making text bold: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/43ca887f-3c01-4735-9fa8-1a6677deb4d1)

**Comment by ilmartyrk** [comment]
est. dev 15. min fix

**Comment by ssin1901** [comment]
@BeccaMelhuish this seems to be fixed in Android Chrome (screenshot1). Can I move it to Done?

![IMG_0196](https://github.com/user-attachments/assets/6d1e814f-6fea-427a-a936-6086be6b6055)

**Comment by BeccaMelhuish** [comment]
@ssin1901 @ilmartyrk afraid it's not fixed for me on desktop, Windows, Chrome:

My attempt at a bold heading followed by 2 bullet points, the second of which bold: 

![image](https://github.com/user-attachments/assets/52a6d8fa-e24d-4be7-a230-d0109f17748b)

Paragraphs are working now, but my attempt at adding a bold heading did this:

![image](https://github.com/user-attachments/assets/af2339b2-1e7c-4d85-8038-fd06ca8ad0dd)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @ssin1901 I don't have a good fix for that, in some cases it get's broken, but at the same time, users can use the preview and also try to fix the broken formatting.  Will remove the high-priority label as the text can be added and also fixed if it should happen.

**Comment by BeccaMelhuish** [comment]
@ilmartyrk @ssin1901 @kevincrepin I'm adding a summary of what we discussed in the product meeting today (@ilmartyrk please correct me if I misrepresent anything): 

**The problem:**
- In [this issue](https://github.com/citizenos/citizenos-fe/issues/1509), we decided to hide the markdown characters for formatting text in ideas, as it's not very user friendly and can cause confusion for users. 
- However, as per this issue above and also this video [here](https://github.com/citizenos/citizenos-fe/issues/1509#issuecomment-2321441765), it sometimes breaks, and causes "bugginess", which is also not very user-friendly and can cause confusion, not look very professional, etc. This isn't possible to fix, so if we kept the same solution, it would be sometimes buggy for users. 
- When it comes to adding URLs in [this issue](https://github.com/citizenos/citizenos-fe/issues/1510), Ilmar can add the UX as per Kevin's design, but in the text itself it will still have to show up as written in markdown.    
- It's not possible to use Etherpad for the ideas, as it's too heavy (simplistic explanation, Ilmar can explain better)  

**Two sides we discussed when considering how to solve the problem**:

1. One option is to remove the possibility for formatting and URLs altogether. The argument for this being that using markdown isn't very user-friendly for the average user, and isn't a good fit with the rest of the UX/UI, if we want to keep things very simple. 

2. Another option is to go back to having the idea text display in markdown when the users are typing, i.e. when you click on the button to make something bold, it changes to **bold** in the text, rather than actually displaying as bold. Same for italics, bullets, URLs, etc. And we would have a 'Preview' button so that users can toggle between the markdown view and how it will look when published. So basically, the same UX as Github has. 

**Decision but depending on @kevincrepin's thoughts.**  

We decided that we would go ahead with option 2, unless @kevincrepin vetoes it and would prefer we just remove formatting and adding URLs entirely, to avoid having complicated UX. 

Basically, both options are pretty bad for the user experience, but after discussion we decided point 2 was the best option out of them.

**Design needed if @kevincrepin agrees to go ahead with option 2:** 

If we go ahead with this, @kevincrepin would you be able to make the design for how users should switch below markdown view and preview view?

**Comment by kevincrepin** [comment]
I agree option 2 makes more sense because for people who do want it at least we offer it then.

And I think GH here does it pretty well so no need to reinvent the wheel :)
![Screenshot 2024-09-12 at 15 58 16](https://github.com/user-attachments/assets/0c9ba98c-6ee7-45d9-91c0-d7353d38478a)

@ilmartyrk component can be found here: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=47-12125&t=SR2BhFrX9LMBZkE3-1

**Comment by BeccaMelhuish** [comment]
@kevincrepin what do you think if we add some sort of explainer message here for when people are adding a link, so it's a bit less surprising/confusing when it appears in markdown?  

![image](https://github.com/user-attachments/assets/8a525a7b-8184-4772-92f7-7126c04113bd)

As they'd likely be expecting it to just look like a normal link, but then suddenly it looks like this and they're possibly thinking something's gone wrong: 

![image](https://github.com/user-attachments/assets/f4ff6519-dfdd-4f95-a1f9-46b6921aa9e4)

With the formatting it's a bit less unexpected, but for the link it seems more so, so perhaps the add link pop-up is a good place to explain?

**Comment by kevincrepin** [comment]
Sure, we can have a little explainer there:

![Screenshot 2024-09-13 at 15 46 17](https://github.com/user-attachments/assets/5f7dd0bf-f388-477a-a735-f8dee12575bc)

**Comment by BeccaMelhuish** [comment]
@kevincrepin Nice! What about 'Don't worry it'll look better published, check the preview :)' for the second line though? 

And I guess most users wouldn't know what 'markdown' means, but might Google it and then learn? (Personally I've only just learned this term). 

Or we could add a tooltip here, explaining what markdown is, and how to use it? We discussed in our product meeting that a tooltip about markdown would be useful for this solution, but this seems the only suitable place for it?

**Comment by kevincrepin** [comment]
Maybe we could say "in the text field" or something? I don't know how I feel about also explaining markdown now to users :) Is that the point of the platform to explain that? :) I guess it doesn't serve much benefit except in this case where we would use the term. So maybe we can avoid it altogether?

![Screenshot 2024-09-13 at 16 05 13](https://github.com/user-attachments/assets/0db9905f-ea8c-45c9-8c07-cde7989bb6b4)

**Comment by BeccaMelhuish** [comment]
OK yes I think this is good, and then we can see when we next have user tests if there's any need for more guidance or not :)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk @kevincrepin checked it and just a couple of things:

In the 'Write' tab, it still makes the text bold, should it? Or that should only be shown in the Preview now I think? 

![image](https://github.com/user-attachments/assets/45dd2334-941e-432b-a0bc-2eabffe77e13)

Will also point out some bugs for @ilmartyrk re ways it isn't working in preview, though I understand the whole reason we now have this preview tab is because it would sometimes break in unfixable ways (so now the user can fix manually if this happens). But if possible to make them work automatically, would be better :) 

Bold bullet points don't work as then it's 3 bullets in a row (I see GH uses a hyphen for bullets instead, is that an option)? 

![image](https://github.com/user-attachments/assets/a08a5ca3-c6e7-4cd1-8b56-9cc1ded5f4e1)

When I tried to make this 'Tune in from' paragraph bold it put the second set of ** in the wrong place, which did different things to the formatting in Write and Preview, but neither as they should be: 

![image](https://github.com/user-attachments/assets/37ebb8d6-5f16-4320-b4f1-e40702c1d4db)

![image](https://github.com/user-attachments/assets/36544397-ebb8-4714-8383-4b81ea003297)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish I see the issues and I could actually fix them with playing around a little bit, it didn't happen all the time, but it is difficult to narrow them down. I would rather not put any more time into it at least for now as I guess most people would not use bold lists etc.

**Comment by BeccaMelhuish** [comment]
@ilmartyrk OK sure I guess as we're short on time we could leave it for now, and see in user tests (whenever they next may be) and internal team usage how soon any more time should be spent on it :) 

I thought the same re bold bullet lists initially, but on reflection I think they could be quite often used - as people highlight certain points within them, like these below (I know I often do myself for structure):
![image](https://github.com/user-attachments/assets/4d364cdc-7ca5-4375-8c78-05be302501db)
![image](https://github.com/user-attachments/assets/3cb77398-f036-4211-9321-54c56ad9fc22)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish I tested it and bold lists (bullet and numbered) both worked for me, I played around a little and in some cases it did break, but I played around a little, removed the formatting and tried adding it around text where I wanted and got it fixed too. So my take is that it does work, but I agree it takes some effort to get it right in cases

**Comment by BeccaMelhuish** [comment]
OK I will close this for now and move to testing for @ssin1901, just to make sure there aren't any issues we've missed so far, but otherwise I guess we'll just keep it as it is as the best we can do for now, and hopefully finetune when there's more developer time :)

**Comment by ssin1901** [comment]
@BeccaMelhuish @ilmartyrk I encountered some difficulties with italicisation (#1600 ) but it worked with another idea. 

I think with some playing around, users can get it right. Can I move to Done?

**Comment by BeccaMelhuish** [comment]
I will move it to UX research column, but keep it closed :)
